From laurent.pinchart@ideasonboard.com  Fri Feb 12 11:37:13 2010
From: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
Date: Wed, 10 Feb 2010 13:32:49 +0100
Subject: class: Free the class private data in class_release
To: gregkh@suse.de
Cc: kay.sievers@vrfy.org, linux-kernel@vger.kernel.org, artem.bityutskiy@nokia.com
Message-ID: <1265805169-31315-1-git-send-email-laurent.pinchart@ideasonboard.com>


Fix a memory leak by freeing the memory allocated in __class_register
for the class private data.

Signed-off-by: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
Acked-by: Artem Bityutskiy <Artem.Bityutskiy@nokia.com>
Cc: stable <stable@kernel.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/base/class.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/base/class.c
+++ b/drivers/base/class.c
@@ -59,6 +59,8 @@ static void class_release(struct kobject
 	else
 		pr_debug("class '%s' does not have a release() function, "
 			 "be careful\n", class->name);
+
+	kfree(cp);
 }
 
 static struct sysfs_ops class_sysfs_ops = {
